src/hooks/imagePicker.ts

```diff
@@ -77,10 +86,38 @@ export const useCaptureImage = (
     });
 
     if (!result.canceled) {
-      onChange ? onChange(setImageUri)(result.assets[0].uri) : setImageUri(result.assets[0].uri);
+      const uri = result.assets[0].uri;
+      onChange ? onChange(setImageUri)(uri) : setImageUri(uri);
+
+      if (saveImage) {
+        await saveImageToGallery(uri);
+      }
+
       return result.assets[0];
     }
   }, [onChange]);
 
+  const saveImageToGallery = async (uri: string) => {
```

Comment:
this does not need to be part of the hook and would be nice to be defined before used for better readability and understanding.